rspec-mode.el

```diff
@@ -781,12 +791,18 @@ or a cons (FILE . LINE), to run one example."
       (rvm-activate-corresponding-ruby))
 
   (let ((default-directory (or (rspec-project-root) default-directory)))
-    (compile (rspec--vagrant-wrapper
-              (rspec--docker-wrapper
-               (mapconcat 'identity `(,(rspec-runner)
-                                     ,(rspec-runner-options opts)
-                                     ,target) " ")))
-             'rspec-compilation-mode)))
+    (compile
+     (rspec-compile-command target opts)
+     'rspec-compilation-mode)))
+
+(defun rspec-compile-command (target &optional opts)
+  "Composes RSpec command line for the compile function"
+  (let ((default-directory (or (rspec-project-root) default-directory)))
```

Comment:
I don't like this construct already, and here you are adding a second place it's used. Even though one caller of this new function is the other place.

Can we do it in some different way?